### PR TITLE
Be posts CRUD

### PR DESCRIPTION
**Pull request by Wozniaxos**

### CODE REVIEW

File: backend/lib/til/shareable_content.ex [hunk 1]
```diff
@@ -0,0 +1,48 @@
+defmodule Til.ShareableContent do
+  import Ecto.Query, warn: false
+  alias Til.Repo
+  alias Til.ShareableContent.{Post, Category}
+
+  def get_posts, do: Repo.all(Post) |> preload_post_data
```
Comment (qbart) [inline backend/lib/til/shareable_content.ex]: always use `()` at the end of functions

File: backend/lib/til_web/controllers/category_controller.ex [hunk 2]
```diff
@@ -0,0 +1,11 @@
+defmodule TilWeb.CategoryController do
+  use TilWeb, :controller
+  alias Til.Accounts
+  alias Til.ShareableContent
+
+  def index(conn, _) do
+    conn
+      |> put_status(:ok)
+      |> render("index.json", categories: ShareableContent.get_categories)
```
Comment (qbart) [inline backend/lib/til_web/controllers/category_controller.ex]:
```suggestion
      |> render("index.json", categories: ShareableContent.get_categories())
```

File: backend/lib/til_web/views/post_view.ex [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  alias Til.Repo
+
+  def render("index.json", %{posts: posts}) do
+    posts
+    |> Enum.map(&serialize_post/1)
+  end
+
+  def render("show.json", %{post: post}) do
+    post
+    |> serialize_post
```
Comment (qbart) [inline backend/lib/til_web/views/post_view.ex]:
```suggestion
    |> serialize_post()
```

File: backend/priv/repo/migrations/20200318143121_create_posts.exs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+defmodule Til.Repo.Migrations.CreatePosts do
+  use Ecto.Migration
+
+  def change do
+    create table(:posts) do
+      add :title, :string, null: false
+      add :body, :text
+      add :is_public, :boolean
```
Comment (qbart) [inline backend/priv/repo/migrations/20200318143121_create_posts.exs]:
```suggestion
      add :is_public, :boolean, null: false, default: false
```

Comment (qbart) [reply backend/priv/repo/migrations/20200318143121_create_posts.exs]: never do 3-state booleans (null, false, true) :)

File: backend/priv/repo/migrations/20200318143121_create_posts.exs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      add :likes_count, :integer
```
Comment (qbart) [inline backend/priv/repo/migrations/20200318143121_create_posts.exs]:
```suggestion
      add :likes_count, :integer, null: false, default: 0
```

File: backend/priv/repo/seeds.exs [hunk 6]
```diff
@@ -1,11 +1,30 @@
...
-#     mix run priv/repo/seeds.exs
-#
-# Inside the script, you can read and write to any of your
-# repositories directly:
-#
-#     Til.Repo.insert!(%Til.SomeSchema{})
-#
-# We recommend using the bang functions (`insert!`, `update!`
-# and so on) as they will fail if something goes wrong.
+alias Til.ShareableContent.Category
```
Comment (qbart) [inline backend/priv/repo/seeds.exs]: in the future we will need to migrate all categories

File: backend/test/til_web/controllers/post_controller_test.exs [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+          title: post_title,
+          body: post_body,
+          categoriesIds: []
+        })
+
+      assert response.status == 201
+
+      {:ok, parsed_response_body} = Jason.decode(response.resp_body)
+
+      created_post = ShareableContent.get_post_by(title: post_title)
```
Comment (qbart) [inline backend/test/til_web/controllers/post_controller_test.exs]: not sure if we should use our module to fetch data in tests, as they can contain specific logic
@Wozniaxos @mikoscz thoughts?

File: backend/lib/til/shareable_content/post.ex [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  alias Til.Accounts.User
+  alias Til.ShareableContent.Category
+
+  schema "posts" do
+    field :title, :string
+    field :body, :string
+    field :is_public, :boolean
+    field :likes_count, :integer
+
+    belongs_to(:author, User)
```
Comment (qbart) [inline backend/lib/til/shareable_content/post.ex]: in DSL-like scenarios it's ok to skip parentheses

File: backend/test/til_web/controllers/post_controller_test.exs (hunk 7 above)
Comment (Wozniaxos) [reply backend/test/til_web/controllers/post_controller_test.exs]: Yeah, you right, during some additional operation tests will be less efficient and this is also coupled with module.

**Pull request merged**